src/main/java/com/zufar/icedlatte/common/validation/email/EmailValidator.java

```diff
@@ -1,19 +1,22 @@
 package com.zufar.icedlatte.common.validation.email;
 
 import com.zufar.icedlatte.user.repository.UserRepository;
-import jakarta.validation.ConstraintValidator;
-import jakarta.validation.ConstraintValidatorContext;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 
 @RequiredArgsConstructor
 @Component
-public class UniqueEmailValidator implements ConstraintValidator<UniqueEmail, String> {
+public class EmailValidator {
+
+    private static final String emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
 
     private final UserRepository userCrudRepository;
 
-    @Override
-    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+    public boolean isNotValid(String email) {
+        return email == null || !email.matches(emailRegex);
+    }
+
```

Comment:
nit: I prefer `Pattern` over `String.matches`, because:

> String.matches(): This method internally compiles the regex pattern into a Pattern object every time it’s called. If you use it repeatedly with the same regex, the regex is compiled each time, which can be inefficient.
Pattern: When you compile a regex into a Pattern object using Pattern.compile(), it’s compiled once and can be reused multiple times. This avoids the overhead of recompiling the pattern for each match attempt.

Refactored code would be

```java
private static final Pattern EMAIL_PATTERN = Pattern.compile("^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$");
public boolean isNotValid(String email) {
     return email == null || !EMAIL_PATTERN.matcher(email).matches();
}
```

Reply:
UPD: Haven't noticed initially, that you delegated that to Open API, got it 👍